server/src/com/thoughtworks/go/server/scheduling/BuildCauseProducerService.java

```diff
@@ -246,6 +251,46 @@ ServerHealthState newProduceBuildCause(PipelineConfig pipelineConfig, BuildType
         }
     }
 
+    private boolean isGoodReasonToSchedule(PipelineConfig pipelineConfig, BuildCause buildCause, BuildType buildType,
+                                           boolean materialConfigurationChanged,MaterialRevisions peggedRevisions) {
+        if(buildCause == null)
+            return false;
+
+        boolean validCause = buildType.isValidBuildCause(pipelineConfig, buildCause);
+
```

Comment:
> ignoring materialConfigurationChanged flag when pipelineConfig.isConfigOriginSameAsOneOfMaterials() == true

When configuration and pipeline material are in the same repository and there is **some** build cause then it can be either manual trigger or a change in one of the materials. The behavior we want is to run build always when manually forced or build a *scm-config consistent* revision. If revision changed then config has changed.
What you have written is the same as
```
 if (buildCause.isForced() || buildCause.pipelineConfigAndMaterialRevisionMatch(pipelineConfig)) {
                return false;
            }
```
I guess you meant `return true`
But this is not desired.

I'd like to stick with:
```
            if(buildCause.isForced())
            {
                // build is manual - skip scm-config consistency
                return validCause;
            }
            // then we need config and material revisions to be consistent
            if(!buildCause.pipelineConfigAndMaterialRevisionMatch(pipelineConfig))
            {
                return false;
            }

            return validCause;
```
Let me know if it is not OK.